### PR TITLE
Refactor search logic out of views.py

### PR DESCRIPTION
**Pull request by yeraydiazdiaz**
Split from #5311

### CODE REVIEW

**Comment by yeraydiazdiaz** [comment]
The motivation was to avoid having the actual fields in the querystring and in the template, as this exposes internal details and couples the template and the searching logic.

I didn't really think about ordering to be honest. Maybe we should officially introduce it as a second dropdown? Though I'm not sure if they would apply to the score-based modes if you find "desc created" useful I'm sure more people would.

**Comment by nlhkabu** [comment]
Note:
If / when this PR is re-reviewed, we will need to ensure that the template is appropriately translated.

**Comment by yeraydiazdiaz** [comment]
I've updated this MR taking @di's comments into account and is ready for review.

I verified the search results are the same both in master and this branch.

**Comment by yeraydiazdiaz** [comment]
I forgot to mention re @nlhkabu's comment above, there are no template changes anymore so I believe no translation changes are required.

File: warehouse/templates/search/results.html [hunk 1]
```diff
@@ -178,9 +178,9 @@ <h2 class="sr-only">{% trans %}Search results{% endtrans %}</h2>
             <input id="search" type="hidden" name="q" value="{{ term }}">
             <label for="order">{% trans %}Order by{% endtrans %} &nbsp;</label>
             <select class="-js-form-submit-trigger" id="order" name="o">
-              {{ search_option(gettext("Relevance"), "") }}
-              {{ search_option(gettext("Date last updated"), "-created") }}
-              {{ search_option(gettext("Trending"), "-zscore") }}
+              {{ search_option("Relevance", "") }}
```
Comment (ewdurbin) [inline warehouse/templates/search/results.html]: will removing the `gettext` "untranslated" these strings?

Comment (yeraydiazdiaz) [reply warehouse/templates/search/results.html]: Woops, yeah, that wasn't intentional, good catch 👍

**Pull request merged**